src/Core/functions.php

```diff
@@ -206,11 +206,10 @@ function env(string $key, string|float|int|bool|null $default = null): string|fl
             $key = 'SCRIPT_URL';
         }
 
-        /** @var string|null $val */
         $val = $_SERVER[$key] ?? $_ENV[$key] ?? null;
+        assert($val === null || is_scalar($val));
         if ($val == null && getenv($key) !== false) {
-            /** @var string|false $val */
-            $val = getenv($key);
+            $val = (string)getenv($key);
```

Comment:
How are we not testing this somehow?

Reply:
We probably haven't run into any of the cases that would trigger the value to not be a string.